packages/clarity-js/src/interaction/encode.ts

```diff
@@ -117,13 +117,17 @@ export default async function (type: Event, ts: number = null): Promise<void> {
         case Event.Scroll:
             for (let entry of scroll.state) {
                 let sTarget = metadata(entry.data.target as Node, entry.event);
+                const top = metadata(entry.data.top as Node, entry.event);
+                const bottom = metadata(entry.data.bottom as Node, entry.event);
+                const cTopHash = top.hash ? top.hash.join(Constant.Dot) : Constant.Empty;
```

Comment:
>top [](http://example.com/codeflow?start=33&length=3)

if there is any risk of top or bottom being null you may want to do top?.hash and bottom?.

Reply:
metadata returns a object by default so top or bottom won't be null!